tests/integration/connector.py

```diff
@@ -36,3 +37,55 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.connection.commit()
         self.cursor.close()
         self.connection.close()
+
+
+class PostgreSQLConnector:
+    """Context manager for PostrgreSQL connector."""
+
+    def __init__(self, config: str):
+        """Initialize the context manager.
+
+        Args:
+            config: Connection string for the postgresql connector
+        """
+        self.config = config
+
+    def __enter__(self):
+        """Create the connection and return a cursor."""
+        self.connection = psycopg2.connect(self.config)
+        self.connection.autocommit = True
+        self.cursor = self.connection.cursor()
+        return self.cursor
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Handle connection close."""
+        self.cursor.close()
+        self.connection.close()
+
+
+class MongoDBConnector:
+    """Context manager for PostrgreSQL connector."""
+
+    def __init__(self, config: str):
+        """Initialize the context manager.
+
+        Args:
+            config: Configuration uri for the mongodb connector
+        """
+        self.config = config
+
+    def __enter__(self):
```

Comment:
Also in MongoDB one can use unit-of-work, using sessions and transactions, via contexts 

https://pymongo.readthedocs.io/en/stable/api/pymongo/client_session.html